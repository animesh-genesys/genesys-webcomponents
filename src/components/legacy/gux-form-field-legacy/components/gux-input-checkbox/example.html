<form onchange="notify(event)" oninput="notify(event)">
  <h1>gux-form-field-legacy (Checkbox)</h1>

  <p>
    slot a input[type=checkbox] element into a gux-form-field-legacy element
  </p>

  <div>
    <gux-form-field-legacy>
      <input slot="input" type="checkbox" name="food" value="pizza" />
      <label slot="label">Pizza</label>
    </gux-form-field-legacy>
    <gux-form-field-legacy>
      <input slot="input" type="checkbox" name="food" value="pasta" checked />
      <label slot="label">Pasta</label>
    </gux-form-field-legacy>
    <gux-form-field-legacy>
      <input
        slot="input"
        type="checkbox"
        name="food"
        value="hamburger"
        id="hamburger"
      />
      <label slot="label">Hamburger</label>
    </gux-form-field-legacy>
    <gux-form-field-legacy>
      <input
        slot="input"
        type="checkbox"
        name="food"
        value="sandwich"
        disabled
      />
      <label slot="label">Sandwich</label>
    </gux-form-field-legacy>
    <gux-form-field-legacy>
      <input slot="input" type="checkbox" name="food" value="sushi" />
      <label slot="label">Sushi</label>
      <span slot="error">Required</span>
    </gux-form-field-legacy>
  </div>
</form>

<style
  onload="(function () {
  document.getElementById('hamburger').indeterminate = true;
})()"
>
  .not-used {
    -custom-noop: noop;
  }
</style>
