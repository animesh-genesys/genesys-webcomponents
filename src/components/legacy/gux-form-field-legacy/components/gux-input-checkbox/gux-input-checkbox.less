@import (reference) '../../../../../style/color.less';

@gux-icon-checkbox-unchecked: url("data:image/svg+xml, %3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'%3E%3Cpath d='M11.138 2.467a2 2 0 0 1 2 2v7a2 2 0 0 1-2 2h-7a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h7m0-1h-7a3 3 0 0 0-3 3v7a3 3 0 0 0 3 3h7a3 3 0 0 0 3-3v-7a3 3 0 0 0-3-3z' fill='%2377828f' /%3E%3C/svg%3E");
@gux-icon-checkbox-unchecked-hover: url("data:image/svg+xml, %3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'%3E%3Cpath d='M11.138 2.467a2 2 0 0 1 2 2v7a2 2 0 0 1-2 2h-7a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h7m0-1h-7a3 3 0 0 0-3 3v7a3 3 0 0 0 3 3h7a3 3 0 0 0 3-3v-7a3 3 0 0 0-3-3z' fill='%232a60c8' /%3E%3C/svg%3E");
@gux-icon-checkbox-unchecked-error: url("data:image/svg+xml, %3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'%3E%3Cpath d='M11.138 2.467a2 2 0 0 1 2 2v7a2 2 0 0 1-2 2h-7a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h7m0-1h-7a3 3 0 0 0-3 3v7a3 3 0 0 0 3 3h7a3 3 0 0 0 3-3v-7a3 3 0 0 0-3-3z' fill='%23ea0b0b' /%3E%3C/svg%3E");
@gux-icon-checkbox-checked: url("data:image/svg+xml, %3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'%3E%3Cpath d='M11.138 1.467h-7a3 3 0 0 0-3 3v7a3 3 0 0 0 3 3h7a3 3 0 0 0 3-3v-7a3 3 0 0 0-3-3zM6.8 11 3.638 7.941l.89-.932L6.8 9.2l3.947-3.861.891.932z' fill='%232a60c8' /%3E%3C/svg%3E");
@gux-icon-checkbox-mixed: url("data:image/svg+xml, %3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'%3E%3Cpath d='M4.138 8.467h7v-1h-7zm7-7h-7a3 3 0 0 0-3 3v7a3 3 0 0 0 3 3h7a3 3 0 0 0 3-3v-7a3 3 0 0 0-3-3zm2 10a2 2 0 0 1-2 2h-7a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h7a2 2 0 0 1 2 2z' fill='%232a60c8' /%3E%3C/svg%3E");

gux-input-checkbox {
  display: block;
  color: @gux-black-50;

  .gux-input-checkbox-container {
    position: relative;
    padding-left: 24px;
    line-height: 24px;
  }

  input {
    position: absolute;
    z-index: -1;
    opacity: 0;
  }

  label {
    display: inline-block;
    font-size: 12px;

    &:hover {
      cursor: pointer;
    }
  }

  label::after {
    position: absolute;
    top: 4px;
    left: 4px;
    display: block;
    width: @gux-icon-size-default-global;
    height: @gux-icon-size-default-global;
    content: '';
    border-radius: 15%;
  }

  input:focus-within ~ label::after {
    box-shadow: 0 0 3px 2px @gux-blue-90;
  }

  input:not(:checked) ~ label::after {
    background-image: @gux-icon-checkbox-unchecked;
  }

  &.gux-input-error input:not(:checked) ~ label::after {
    background-image: @gux-icon-checkbox-unchecked-error;
  }

  input:not(:checked):not(:disabled):not(:indeterminate) ~ label:hover::after {
    background-image: @gux-icon-checkbox-unchecked-hover;
  }

  input:checked ~ label::after {
    background-image: @gux-icon-checkbox-checked;
  }

  input:indeterminate ~ label::after {
    background-image: @gux-icon-checkbox-mixed;
  }

  input:disabled ~ label::after,
  input:disabled ~ label {
    cursor: not-allowed;
    opacity: 0.5;
  }
}
