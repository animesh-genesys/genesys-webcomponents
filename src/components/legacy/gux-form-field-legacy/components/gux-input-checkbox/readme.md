# gux-input-checkbox

<!-- Auto Generated Below -->


## Slots

| Slot      | Description                              |
| --------- | ---------------------------------------- |
| `"input"` | Required slot for input[type="checkbox"] |
| `"label"` | Required slot for label                  |


## Dependencies

### Used by

 - [gux-form-field-legacy](../..)

### Graph
```mermaid
graph TD;
  gux-form-field-legacy --> gux-input-checkbox
  style gux-input-checkbox fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
