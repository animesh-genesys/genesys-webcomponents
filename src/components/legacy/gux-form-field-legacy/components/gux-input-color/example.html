<form onchange="notify(event)" oninput="notify(event)">
  <h1>gux-form-field-legacy (Color)</h1>

  <p>slot a input[type=color] element into a gux-form-field-legacy element</p>

  <h2>Label Position Above</h2>

  <div>
    <gux-form-field-legacy label-position="above">
      <input slot="input" name="color" type="color" value="#1da8b3" />
      <label slot="label">Color</label>
    </gux-form-field-legacy>
  </div>

  <div>
    <gux-form-field-legacy label-position="above">
      <input slot="input" name="color" type="color" value="#1da8b3" disabled />
      <label slot="label">Color</label>
    </gux-form-field-legacy>
  </div>

  <div>
    <gux-form-field-legacy label-position="above">
      <input required slot="input" name="color" type="color" value="#1da8b3" />
      <label slot="label">Color </label>
      <span slot="error">Required</span>
    </gux-form-field-legacy>
  </div>

  <h2>Label Position Beside</h2>

  <div>
    <gux-form-field-legacy label-position="beside">
      <input slot="input" name="color" type="color" value="#1da8b3" />
      <label slot="label">Color</label>
    </gux-form-field-legacy>
  </div>

  <div>
    <gux-form-field-legacy label-position="beside">
      <input slot="input" name="color" type="color" value="#1da8b3" disabled />
      <label slot="label">Color</label>
    </gux-form-field-legacy>
  </div>

  <div>
    <gux-form-field-legacy label-position="beside">
      <input required slot="input" name="color" type="color" value="#1da8b3" />
      <label slot="label">Color</label>
      <span slot="error">Required</span>
    </gux-form-field-legacy>
  </div>
</form>
