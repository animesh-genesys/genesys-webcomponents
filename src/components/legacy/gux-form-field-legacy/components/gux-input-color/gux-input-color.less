@import (reference) '../../../../../style/color.less';
@import (reference) '../../../../../style/typography.less';

gux-input-color {
  color: @gux-black-50;

  > section {
    position: relative;
    display: inline-block;
    width: 160px;
  }

  // Don't render options as direct children of the picker
  // They'll render a slot components when open
  > gux-input-color-option {
    display: none;
  }

  .gux-input-color-main-element {
    box-sizing: border-box;
    width: 100%;
    padding: 0;
    margin: 0;
    cursor: pointer;
    user-select: none;
    background-color: @gux-grey-90;
    border: 1px solid @gux-grey-30;
    border-radius: 4px;
    outline: none;
    box-shadow: 0 0 0 0 transparent;
    box-shadow: inset 0 0 4px fade(@gux-black-30, 16%);
    transition: box-shadow 0.2s;

    &:focus,
    &.gux-opened {
      border-color: @gux-blue-60;
      outline: none;
      box-shadow: 0 0 4px fade(@gux-blue-80, 50%);
    }

    &:disabled {
      pointer-events: none;
      cursor: default;
      opacity: 0.5;
    }

    &:hover,
    &:focus {
      gux-icon {
        color: @gux-black-50;
      }
    }
  }

  .gux-input-color-selected-color {
    float: left;
    padding: 8px;
    margin: 8px 12px;
  }

  .gux-input-color-color-name {
    float: left;
    margin: 9px 0;
    font-size: 12px;
  }

  .gux-input-color-color-select {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1;
    display: none;
    width: 100%;

    &.gux-opened {
      display: inherit;
    }
  }

  gux-icon {
    position: absolute;
    top: 0;
    right: 0;
    display: flex;
    align-items: center;
    width: 16px;
    height: 100%;
    padding: 0 7px;
    overflow: hidden;
    color: @gux-black-90;
    pointer-events: none;
    cursor: pointer;
    background: none;
    border: none;
    outline: none;
  }

  .gux-hidden {
    .gux-sr-only();
  }
}
