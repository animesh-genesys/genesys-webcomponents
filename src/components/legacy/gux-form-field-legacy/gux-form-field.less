@import (reference) '../../../style/color.less';
@import (reference) '../../../style/typography.less';

// Variables part
@gux-form-beside-label-container-width: 45px;
@gux-form-beside-label-margin-right: 8px;

// Style
gux-form-field-legacy {
  display: block;
  color: @gux-black-50;

  .gux-label-and-input-and-error-container {
    margin: 16px 0;

    label {
      font-size: 12px;
      font-weight: bold;
      line-height: 20px;
    }

    input,
    select {
      width: 100%;
      font-size: 12px;
    }

    .gux-input-container input {
      overflow: hidden;
      text-overflow: ellipsis;
    }

    .gux-input-error .gux-input-container {
      border-color: @gux-alert;
    }

    .gux-input-error textarea {
      border-color: @gux-alert;
    }

    .gux-input-error .gux-input-color-main-element {
      border-color: @gux-alert;
    }

    .gux-input-and-error-container {
      flex-grow: 1;
    }

    &.gux-beside {
      display: flex;
      flex-direction: row;

      .gux-label-container {
        position: relative;
        top: 7px;
        width: fit-content;
        min-width: @gux-form-beside-label-container-width;
        margin-right: @gux-form-beside-label-margin-right;
      }

      gux-input-range,
      gux-input-select,
      gux-input-text-like,
      gux-input-search {
        flex: 1 1 auto;
      }
    }

    &.gux-screenreader {
      label {
        .gux-sr-only();
      }
    }
  }

  .gux-required::after {
    font-size: 12px;
    color: @gux-alert-red-60;
    content: ' *';
  }

  .gux-error {
    display: none;
    margin: 4px 0;

    &.gux-show {
      display: block;
    }
  }
}
