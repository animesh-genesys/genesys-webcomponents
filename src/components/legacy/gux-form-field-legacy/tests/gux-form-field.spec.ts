import { newSpecPage } from '@stencil/core/testing';
import MutationObserver from 'mutation-observer';

import { GuxInputCheckbox } from 'components/legacy/gux-form-field-legacy/components/gux-input-checkbox/gux-input-checkbox';
import { GuxInputColor } from 'components/legacy/gux-form-field-legacy/components/gux-input-color/gux-input-color';
import { GuxInputRadio } from 'components/legacy/gux-form-field-legacy/components/gux-input-radio/gux-input-radio';
import { GuxInputRange } from 'components/legacy/gux-form-field-legacy/components/gux-input-range/gux-input-range';
import { GuxInputTextLike } from 'components/legacy/gux-form-field-legacy/components/gux-input-text-like/gux-input-text-like';

import { GuxFormFieldLegacy } from '../gux-form-field';

const components = [
  GuxFormFieldLegacy,
  GuxInputCheckbox,
  GuxInputColor,
  GuxInputRadio,
  GuxInputRange,
  GuxInputTextLike
];
const language = 'en';

describe('gux-form-field-legacy', () => {
  beforeEach(async () => {
    global.MutationObserver = MutationObserver;
    jest.spyOn(global.Math, 'random').mockReturnValue(0.5);
  });

  it('should build', async () => {
    const html = `
      <gux-form-field-legacy>
        <input slot="input" type="unknown">
        <label slot="label">Text</label>
      </gux-form-field-legacy>
    `;
    const page = await newSpecPage({ components, html, language });

    expect(page.rootInstance).toBeInstanceOf(GuxFormFieldLegacy);
  });

  describe('#render', () => {
    describe('input tag', () => {
      [
        'checkbox',
        'color',
        'email',
        'number',
        'password',
        'radio',
        'range',
        'search',
        'text',
        'unknown'
      ].forEach(type => {
        it(`should render component type "${type}"`, async () => {
          const html = `
            <gux-form-field-legacy>
              <input
                slot="input"
                type="${type}"
                value="test"
              />
              <label slot="label">Test</label>
            </gux-form-field-legacy>
          `;
          const page = await newSpecPage({ components, html, language });

          expect(page.root).toMatchSnapshot();
        });
        it(`should render component type "${type}" with an error slot`, async () => {
          const html = `
            <gux-form-field-legacy>
              <input
                slot="input"
                type="${type}"
                value="test"
              />
              <label slot="label">Test</label>
              <span slot="error">Error Message</span>
            </gux-form-field-legacy>
          `;
          const page = await newSpecPage({ components, html, language });

          expect(page.root).toMatchSnapshot();
        });
      });
    });

    describe('select tag', () => {
      it(`should render component type "select"`, async () => {
        const html = `
          <gux-form-field-legacy>
            <select slot="input" name="select">
              <option value="option1">Option 1</option>
              <option value="option1">Option 2</option>
              <option value="option1">Option 3</option>
            </select>
            <label slot="label">Select</label>
          </gux-form-field-legacy>
        `;
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });

      it(`should render component type "select" with an error slot`, async () => {
        const html = `
          <gux-form-field-legacy>
            <select slot="input" name="select">
              <option value="option1">Option 1</option>
              <option value="option1">Option 2</option>
              <option value="option1">Option 3</option>
            </select>
            <label slot="label">Select</label>
            <span slot="error">Error Message</span>
          </gux-form-field-legacy>
        `;
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });

      it(`should render component type "select" and keep user provided id and for attributes`, async () => {
        const html = `
          <gux-form-field-legacy>
            <select slot="input" id="test" name="select">
              <option value="option1">Option 1</option>
              <option value="option1">Option 2</option>
              <option value="option1">Option 3</option>
            </select>
            <label for="test" slot="label">Select</label>
            <span slot="error">Error Message</span>
          </gux-form-field-legacy>
        `;
        const page = await newSpecPage({ components, html, language });
        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
