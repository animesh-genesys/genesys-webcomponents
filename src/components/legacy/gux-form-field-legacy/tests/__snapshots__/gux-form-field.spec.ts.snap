// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-form-field-legacy #render input tag should render component type "checkbox" 1`] = `
<gux-form-field-legacy>
  <!---->
  <div>
    <gux-input-checkbox>
      <!---->
      <div class="gux-input-checkbox-container">
        <input id="gux-form-field-i" slot="input" type="checkbox" value="test">
        <label for="gux-form-field-i" slot="label">
          Test
        </label>
      </div>
    </gux-input-checkbox>
    <div class="gux-error" id="gux-form-field-error-i">
      <gux-error-message-beta></gux-error-message-beta>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "checkbox" with an error slot 1`] = `
<gux-form-field-legacy>
  <!---->
  <div>
    <gux-input-checkbox>
      <!---->
      <div class="gux-input-checkbox-container">
        <input aria-describedby="gux-form-field-error-i" id="gux-form-field-i" slot="input" type="checkbox" value="test">
        <label for="gux-form-field-i" slot="label">
          Test
        </label>
      </div>
    </gux-input-checkbox>
    <div class="gux-error" id="gux-form-field-error-i">
      <gux-error-message-beta>
        <span slot="error">
          Error Message
        </span>
      </gux-error-message-beta>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "color" 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label id="gux-form-field-label-i" slot="label">
        Test
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-color gux-error-describedby="gux-form-field-error-i" gux-label-describedby="gux-form-field-label-i">
        <!---->
        <section>
          <span class="gux-hidden" id="gux-input-color-required-i">
            Required
          </span>
          <button aria-describedby="gux-form-field-label-i  gux-form-field-error-i" aria-expanded="false" class="gux-input-color-main-element" type="button">
            <div class="gux-input-color-selected-color" style="background: test;"></div>
            <div class="gux-input-color-color-name">
              test
            </div>
            <gux-icon decorative="" icon-name="chevron-small-down"></gux-icon>
          </button>
          <gux-color-select class="gux-input-color-color-select">
            <input slot="input" type="color" value="test">
          </gux-color-select>
        </section>
      </gux-input-color>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta></gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "color" with an error slot 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label id="gux-form-field-label-i" slot="label">
        Test
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-color gux-error-describedby="gux-form-field-error-i" gux-label-describedby="gux-form-field-label-i">
        <!---->
        <section>
          <span class="gux-hidden" id="gux-input-color-required-i">
            Required
          </span>
          <button aria-describedby="gux-form-field-label-i  gux-form-field-error-i" aria-expanded="false" class="gux-input-color-main-element" type="button">
            <div class="gux-input-color-selected-color" style="background: test;"></div>
            <div class="gux-input-color-color-name">
              test
            </div>
            <gux-icon decorative="" icon-name="chevron-small-down"></gux-icon>
          </button>
          <gux-color-select class="gux-input-color-color-select">
            <input slot="input" type="color" value="test">
          </gux-color-select>
        </section>
      </gux-input-color>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta>
          <span slot="error">
            Error Message
          </span>
        </gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "email" 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-text-like slot="input">
        <!---->
        <div class="gux-input-container">
          <input id="gux-form-field-i" slot="input" type="email" value="test">
        </div>
      </gux-input-text-like>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta></gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "email" with an error slot 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-text-like slot="input">
        <!---->
        <div class="gux-input-container">
          <input aria-describedby="gux-form-field-error-i" id="gux-form-field-i" slot="input" type="email" value="test">
        </div>
      </gux-input-text-like>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta>
          <span slot="error">
            Error Message
          </span>
        </gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "number" 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-number slot="input">
        <input id="gux-form-field-i" slot="input" type="number" value="test">
      </gux-input-number>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta></gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "number" with an error slot 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-number slot="input">
        <input aria-describedby="gux-form-field-error-i" id="gux-form-field-i" slot="input" type="number" value="test">
      </gux-input-number>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta>
          <span slot="error">
            Error Message
          </span>
        </gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "password" 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-text-like slot="input">
        <!---->
        <div class="gux-input-container">
          <input id="gux-form-field-i" slot="input" type="password" value="test">
        </div>
      </gux-input-text-like>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta></gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "password" with an error slot 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-text-like slot="input">
        <!---->
        <div class="gux-input-container">
          <input aria-describedby="gux-form-field-error-i" id="gux-form-field-i" slot="input" type="password" value="test">
        </div>
      </gux-input-text-like>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta>
          <span slot="error">
            Error Message
          </span>
        </gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "radio" 1`] = `
<gux-form-field-legacy>
  <!---->
  <gux-input-radio>
    <!---->
    <div class="gux-input-radio-container">
      <input id="gux-form-field-i" slot="input" type="radio" value="test">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
  </gux-input-radio>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "radio" with an error slot 1`] = `
<gux-form-field-legacy>
  <!---->
  <span hidden="" slot="error">
    Error Message
  </span>
  <gux-input-radio>
    <!---->
    <div class="gux-input-radio-container">
      <input aria-describedby="gux-form-field-error-i" id="gux-form-field-i" slot="input" type="radio" value="test">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
  </gux-input-radio>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "range" 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
    <gux-input-range>
      <!---->
      <div class="gux-container">
        <div class="gux-range">
          <div class="gux-track">
            <div class="gux-progress" style="width: NaN%;"></div>
          </div>
          <input id="gux-form-field-i" slot="input" type="range" value="test">
          <div class="gux-hidden gux-range-tooltip-container">
            <div class="gux-range-tooltip" style="left: NaN%;">
              test
            </div>
          </div>
        </div>
        <div class="gux-display">
          test
        </div>
      </div>
    </gux-input-range>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "range" with an error slot 1`] = `
<gux-form-field-legacy>
  <!---->
  <span hidden="" slot="error">
    Error Message
  </span>
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
    <gux-input-range>
      <!---->
      <div class="gux-container">
        <div class="gux-range">
          <div class="gux-track">
            <div class="gux-progress" style="width: NaN%;"></div>
          </div>
          <input aria-describedby="gux-form-field-error-i" id="gux-form-field-i" slot="input" type="range" value="test">
          <div class="gux-hidden gux-range-tooltip-container">
            <div class="gux-range-tooltip" style="left: NaN%;">
              test
            </div>
          </div>
        </div>
        <div class="gux-display">
          test
        </div>
      </div>
    </gux-input-range>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "search" 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-search>
        <input id="gux-form-field-i" slot="input" type="search" value="test">
      </gux-input-search>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta></gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "search" with an error slot 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-search>
        <input aria-describedby="gux-form-field-error-i" id="gux-form-field-i" slot="input" type="search" value="test">
      </gux-input-search>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta>
          <span slot="error">
            Error Message
          </span>
        </gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "text" 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-text-like slot="input">
        <!---->
        <div class="gux-input-container">
          <input id="gux-form-field-i" slot="input" type="text" value="test">
        </div>
      </gux-input-text-like>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta></gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "text" with an error slot 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Test
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-text-like slot="input">
        <!---->
        <div class="gux-input-container">
          <input aria-describedby="gux-form-field-error-i" id="gux-form-field-i" slot="input" type="text" value="test">
        </div>
      </gux-input-text-like>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta>
          <span slot="error">
            Error Message
          </span>
        </gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "unknown" 1`] = `
<gux-form-field-legacy>
  <!---->
  <div>
    <label for="gux-form-field-i" slot="label">
      Test
    </label>
    <input id="gux-form-field-i" slot="input" type="unknown" value="test">
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render input tag should render component type "unknown" with an error slot 1`] = `
<gux-form-field-legacy>
  <!---->
  <div>
    <label for="gux-form-field-i" slot="label">
      Test
    </label>
    <input aria-describedby="gux-form-field-error-i" id="gux-form-field-i" slot="input" type="unknown" value="test">
    <span slot="error">
      Error Message
    </span>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render select tag should render component type "select" 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Select
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-select slot="input">
        <select id="gux-form-field-i" name="select" slot="input">
          <option value="option1">
            Option 1
          </option>
          <option value="option1">
            Option 2
          </option>
          <option value="option1">
            Option 3
          </option>
        </select>
      </gux-input-select>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta></gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render select tag should render component type "select" and keep user provided id and for attributes 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="test" slot="label">
        Select
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-select slot="input">
        <select aria-describedby="gux-form-field-error-i" id="test" name="select" slot="input">
          <option value="option1">
            Option 1
          </option>
          <option value="option1">
            Option 2
          </option>
          <option value="option1">
            Option 3
          </option>
        </select>
      </gux-input-select>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta>
          <span slot="error">
            Error Message
          </span>
        </gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;

exports[`gux-form-field-legacy #render select tag should render component type "select" with an error slot 1`] = `
<gux-form-field-legacy>
  <!---->
  <div class="gux-above gux-label-and-input-and-error-container">
    <div class="gux-label-container">
      <label for="gux-form-field-i" slot="label">
        Select
      </label>
    </div>
    <div class="gux-input-and-error-container">
      <gux-input-select slot="input">
        <select aria-describedby="gux-form-field-error-i" id="gux-form-field-i" name="select" slot="input">
          <option value="option1">
            Option 1
          </option>
          <option value="option1">
            Option 2
          </option>
          <option value="option1">
            Option 3
          </option>
        </select>
      </gux-input-select>
      <div class="gux-error" id="gux-form-field-error-i">
        <gux-error-message-beta>
          <span slot="error">
            Error Message
          </span>
        </gux-error-message-beta>
      </div>
    </div>
  </div>
</gux-form-field-legacy>
`;
